ssl/bio_ssl.c

```diff
@@ -101,14 +102,14 @@ static int ssl_read(BIO *b, char *out, int outl)
 
     BIO_clear_retry_flags(b);
 
-    ret = SSL_read(ssl, out, outl);
+    ret = SSL_read_ex(ssl, out, outl, &readbytes);
 
     switch (SSL_get_error(ssl, ret)) {
     case SSL_ERROR_NONE:
-        if (ret <= 0)
+        if (readbytes == 0)
```

Comment:
Is that even possible?  `SSL_get_error()` will only return `SSL_ERROR_NONE` when `ret > 0`, and that would never happen when only zero bytes have been read, right?